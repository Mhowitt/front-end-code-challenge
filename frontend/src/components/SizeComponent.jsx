import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import ProgressBar from './ProgressBar/ProgressBar'

const SizeComponent = props => {
  return (
    <div>
      <h2>Now tell us what your body type is!</h2>
      {/* <ProgressBar question={props.question}/> */}
      <div className="card-container">
        <div style={{display: 'flex', flexDirection:'row', justifyContent:'space-around'}}>
          <div>Petite</div>
          <div>Average</div>
          <div>Tall</div>
        </div>
        <div>
          <Link to="/results"><button type="submit" className="start-quiz" onClick={props.handleClick} >Submit your results!</button></Link>
        </div>
      </div>
    </div>
  )
}


export default SizeComponent;