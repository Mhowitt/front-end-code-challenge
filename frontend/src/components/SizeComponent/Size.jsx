import React from 'react';
import { Link } from 'react-router-dom';
import './Size.css';

const Size = props => {
  return (
    <div>
      <h2>Now tell us what your body type is!</h2>
      <div className="card-container">
        <div className="size-container">
          <Link to="/results">
            <button
              className="size-button"
              onClick={props.handleClick}
              value="petite"
            >
              Petite
            </button>
          </Link>
          <Link to="/results">
            <button
              className="size-button"
              onClick={props.handleClick}
              value="average"
            >
              Average
            </button>
          </Link>
          <Link to="/results">
            <button
              className="size-button"
              onClick={props.handleClick}
              value="tall"
            >
              Tall
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
};

export default Size;
