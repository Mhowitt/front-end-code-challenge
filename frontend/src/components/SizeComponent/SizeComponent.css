.size-container {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  position: relative;
  top: 50%;
  transform: translate(0, -50%);
}

.size-button {
  align-self: center;
  width: 80px;
  height: 40px;
  border-radius: 10px;
  background-color: #96858f;
  color: rgb(254, 252, 252);
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
}
