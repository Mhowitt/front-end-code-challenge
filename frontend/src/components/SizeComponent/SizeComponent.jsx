import React from 'react';
import { Link } from 'react-router-dom';
import './SizeComponent.css';

const SizeComponent = props => {
  return (
    <div>
      <h2>Now tell us what your body type is!</h2>
      <div className="card-container">
        <div className="size-container">
          <Link to="/results">
            <button className="size-button" onClick={props.handleClick}>
              Petite
            </button>
          </Link>
          <Link to="/results">
            <button className="size-button" onClick={props.handleClick}>
              Average
            </button>
          </Link>
          <Link to="/results">
            <button className="size-button" onClick={props.handleClick}>
              Tall
            </button>
          </Link>
        </div>
        {/* <div>
          <Link to="/results">
            <button
              type="submit"
              className="start-quiz"
              onClick={props.handleClick}
            >
              Submit your results!
            </button>
          </Link>
        </div> */}
      </div>
    </div>
  );
};

export default SizeComponent;
