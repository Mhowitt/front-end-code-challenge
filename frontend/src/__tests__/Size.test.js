import React from 'react'; // for JSX
import { shallow, mount } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import SizeComponent from '../components/SizeComponent/Size.jsx';

describe('<SizeComponent />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = shallow(<SizeComponent />);
  });
  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
