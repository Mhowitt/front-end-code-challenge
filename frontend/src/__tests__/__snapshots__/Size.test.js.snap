// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SizeComponent /> matches snapshot 1`] = `
<div>
  <h2>
    Now tell us what your body type is!
  </h2>
  <div
    className="card-container"
  >
    <div
      className="size-container"
    >
      <Link
        replace={false}
        to="/results"
      >
        <button
          className="size-button"
          value="petite"
        >
          Petite
        </button>
      </Link>
      <Link
        replace={false}
        to="/results"
      >
        <button
          className="size-button"
          value="average"
        >
          Average
        </button>
      </Link>
      <Link
        replace={false}
        to="/results"
      >
        <button
          className="size-button"
          value="tall"
        >
          Tall
        </button>
      </Link>
    </div>
  </div>
</div>
`;
